fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLInjectionValidator.java

```diff
@@ -60,99 +60,72 @@ public static void validateSQLInput(final String sqlSearch) {
         }
-
-        StringTokenizer tokenizer = new StringTokenizer(inputSqlString, " ");
-        while (tokenizer.hasMoreTokens()) {
-            String token = tokenizer.nextToken().trim();
-            if (token.equals("'")) {
-                if (tokenizer.hasMoreElements()) {
-                    String nextToken = tokenizer.nextToken().trim();
-                    if (!nextToken.equals("'")) {
-                        injectionFound = true;
-                        break;
-                    }
-                } else {
-                    injectionFound = true;
-                    break;
-                }
-            }
-            if (token.equals("\"")) {
-                if (tokenizer.hasMoreElements()) {
-                    String nextToken = tokenizer.nextToken().trim();
-                    if (!nextToken.equals("\"")) {
-                        injectionFound = true;
-                        break;
-                    }
-                } else {
-                    injectionFound = true;
-                    break;
-                }
-            } else if (token.indexOf('=') > 0) {
-                StringTokenizer operatorToken = new StringTokenizer(token, "=");
-                String operand = operatorToken.nextToken().trim();
-                if (!operatorToken.hasMoreTokens()) {
-                    injectionFound = true;
-                    break;
-                }
-                String value = operatorToken.nextToken().trim();
-                if (operand.equals(value)) {
-                    injectionFound = true;
-                    break;
-                }
+        String lowerCaseSQL = sqlSearch.toLowerCase().trim();
+        for (String ddl : DDL_COMMANDS) {
+            if (lowerCaseSQL.startsWith(ddl)) {
+                throw new SQLInjectionException();
             }
         }
-        if (injectionFound) {
-            throw new SQLInjectionException();
-        }
 
-        Pattern pattern = Pattern.compile(SQL_PATTERN);
-        Matcher matcher = pattern.matcher(sqlSearch);
-        if (!matcher.matches()) {
-            throw new SQLInjectionException();
+        for (String comments : COMMENTS) {
+            if (lowerCaseSQL.contains(comments)) {
+                throw new SQLInjectionException();
+            }
         }
+
+        // Removing the space before and after '=' operator
+        // String s = " \" OR 1 = 1"; For the cases like this
+        patternMatchSqlInjection(sqlSearch, lowerCaseSQL);
     }
 
-    public static void validateAdhocQuery(final String sqlSearch) {
+    public static void validateDynamicQuery(final String sqlSearch) {
         if (StringUtils.isBlank(sqlSearch)) {
             return;
         }
-        String lowerCaseSQL = sqlSearch.toLowerCase().trim();
+
+        String lowerCaseSQL = sqlSearch.toLowerCase();
         for (String ddl : DDL_COMMANDS) {
-            if (lowerCaseSQL.startsWith(ddl)) {
+            if (ddl.equals(lowerCaseSQL)) {
```

Comment:
I am not sure the "equals" are enough. I think the values cannot contains any of the restricted commands. 
e.g.: what if the "sqlSearch" value is ` 1; DROP TABLE m_loan; SELECT ` ?